lib/linguist/languages.yml

```
@@ -986,6 +986,15 @@ Cap'n Proto:
   - ".capnp"
   ace_mode: text
   language_id: 52
+Carbon:
+  type: programming
+  color: "#222222"
+  extensions:
+  - ".carbon"
+  ace_mode: golang
+  codemirror_mode: go
+  codemirror_mime_type: text/x-go
```

Syntax highlighting on GitHub is specified by TextMate scope (<samp>tm_scope</samp>). So you'll need to include the [relevant field](https://github.com/github-linguist/linguist/blob/d190ad88af555bb2ee213aa64205f2b9a97a077d/lib/linguist/languages.yml#L7468):

```suggestion
  ace_mode: golang
  codemirror_mode: go
  codemirror_mime_type: text/x-go
  tm_scope: source.v
```